### PR TITLE
workflows: new jumpstart workflow

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Merge the `build` (`pull_request`) and `publish` (`workflow_run`) workflows into a
single `jumpstart` (`pull_request_target`) workflow.

The main issue that this solves is that it makes it much easier to use the
actually built commit ID (the merge commit for pull requests, or the head
commit for the main branch) as an identifier for the resulting cached object.
This solves the inconsistency issues we've been seeing.

We are now building arbitrary code in `pull_request_target`.  We manage this
risk in two ways:

  - the build process (including the checking out of the code) takes place
    as a normal user, inside of a container.  It should be impossible to break
    out.

  - even if a container escape is possible, the VM running the build job has a
    severely restricted GITHUB_TOKEN and no access to secrets

Building inside of a container also gets us a newer version of nodejs, which
slightly increases performance.

We drop the old workflows which used to be responsible for this process.

 -  [] Requires building the  sha256 file: #16004

### CODE REVIEW

File: .github/workflows/webpack-jumpstart.yml
Comment (martinpitt) [inline .github/workflows/webpack-jumpstart.yml:58]: Nice that there's an official one now. You figure we don't need to pin this down to a vetted SHA any more?

Related to that, as further fortification we could [change out action settings](https://github.com/cockpit-project/cockpit/settings/actions) to only allow `actions/*` and `octokit/*` actions, so that it becomes harder to slurp in something untrusted.

File: .github/workflows/webpack-jumpstart.yml
Comment (martinpitt) [inline .github/workflows/webpack-jumpstart.yml:68]: So this is blocked on #16004.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Nice! I like this new structure in general, it's easier to understand, and easier to inspect errors. I still need some convincing that using a new deploy key instead of the cockpituous token is worth the extra effort.

File: .github/workflows/webpack-jumpstart.yml
Comment (allisonkarlitskaya) [reply .github/workflows/webpack-jumpstart.yml:18]: I didn't!  I meant to replace this with docker.  Thanks for the catch.

File: .github/workflows/webpack-jumpstart.yml
Comment (allisonkarlitskaya) [reply .github/workflows/webpack-jumpstart.yml:11]: I didn't find a way to get "just no permissions at all" without giving some kind of a dummy item here.

The statuses are reported out-of-band.  This is just what the token on the runner itself can do.

File: .github/workflows/webpack-jumpstart.yml
Comment (allisonkarlitskaya) [reply .github/workflows/webpack-jumpstart.yml:58]: We have a habit of using freely from the `actions/` namespace without pinning.

This "official one" has existed forever.  We couldn't use it before, though, because it only supports downloading an artifact from the same workflow.

File: .github/workflows/webpack-jumpstart.yml
Comment (allisonkarlitskaya) [reply .github/workflows/webpack-jumpstart.yml:68]: eh.  Let's unblock it and proceed here.  I want to land this, and we can do the sha256sum stuff later.

**Comment by allisonkarlitskaya** [comment]
Tested here:
 - https://github.com/allisoninc/cockpit/pull/1
 - `pull_request_target` https://github.com/allisoninc/cockpit/actions/runs/998310344
 - `push` https://github.com/allisoninc/cockpit/actions/runs/998319362

**Comment by allisonkarlitskaya** [comment]
Dropping the branch protection rule to require `build` success, since it doesn't exist anymore.

**Comment by allisonkarlitskaya** [comment]
Likewise, I expect packit to fail across the board, on account of nothing will feed it the jumpstart tarball from this PR, since `build` never runs (and therefore `publish` is never triggered) and the new workflow doesn't run either, since it's not on main.

File: .github/workflows/webpack-jumpstart.yml
Comment (allisonkarlitskaya) [reply .github/workflows/webpack-jumpstart.yml]: Fixed, and fixed.

No.  I really very much want to stop using the cockpituous token for everything.  Ideally, we'll eventually get to a place where we don't need it at all from workflows anymore.

I propose I'll write a script to automate the process of creating deployment key and environment pairs, with matching ssh keys.

Comment (allisonkarlitskaya) [reply .github/workflows/webpack-jumpstart.yml]: https://en.wikipedia.org/wiki/Principle_of_least_privilege says it all, really.

I hate that we have this key that
 - can do everything; and
 - we use from everywhere

and I'm actively trying to work against that.

Comment (allisonkarlitskaya) [reply .github/workflows/webpack-jumpstart.yml]: See also https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#considering-cross-repository-access

which specifically mentions:

"If a personal access token is used, it should be one that was generated for a new account that is only granted access to the specific repositories that are needed for the workflow. Note that this approach is not scalable and should be avoided in favor of alternatives, such as deploy keys."

ie: if you're gonna make special-purpose accounts ,make them *very* special-purpose, but that's too annoying, so really, just use deploy keys, please.

Also note that "we have plans on the roadmap" → https://github.com/github/roadmap/issues/74

**Review by martinpitt — APPROVED** [review]
Thanks! Nice to untangle that from the sha256 work, and this certainly looks good to me. The deploy keys certainly need automation and documentation, and maybe one of these days GitHub will simplify this (it seems to be a rather common use case?)

**Pull request merged**
